docs/tools/testing/unit_testing.md

```diff
@@ -1,27 +1,275 @@
+2. Install python and pip:
+```
+sudo apt-get -y install python python-setuptools
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+#### Installing dependencies on Mac OS
+
+1. Install [Homebrew](https://brew.sh/).
+2. Install gcc compilers and cmake with: `brew install gcc cmake`
+3. Install python and pip:
+```
+brew install python
+sudo easy_install pip
+```
+3. Install [gcovr](#installing-covr).
+4. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+#### Installing dependencies on Windows
+
+1. Download and install [MinGW-W64](http://mingw-w64.org/).
+2. Download CMake binaries from https://cmake.org/download/ and run the installer.
+3. Download Python2.7 or Python3 from https://www.python.org/getit/ and run the installer.
+4. Add MinGW, CMake and Python into PATH.
+5. Install [gcovr](#installing-covr).
+6. (Optional) Install [Mbed CLI](https://os.mbed.com/docs/latest/tools/arm-mbed-cli.html).
+
+#### Installing gcovr
+
+Install gcovr code coverage tool globally with `pip install 'gcovr>=4.1'` or if you need an isolated Python environment use virtualenv:
+
+##### virtualenv
+
+1. Install virtualenv if not installed with `pip install virtualenv`
+2. Install gcovr with:
+
+Debian/Ubuntu/Mac OS:
+```
+virtualenv pyenv
+. pyenv/bin/activate
+pip install 'gcovr>=4.1'
+```
+
+Windows:
+```
+virtualenv pyenv
+pyenv\\Scripts\\activate
+pip install "gcovr>=4.1"
+```
 
-#### Test code structure
+### Test code structure
 
-Unit tests are located in Mbed OS repository under `UNITTESTS`. Each unit test uses an identical directory tree structure to the file to be tested. This makes it easier to find unit tests for a particular class or a module. For example, if the file to be tested is `rtos/Semaphore.cpp`, then all the test files are in the `UNITTESTS/rtos/Semaphore` directory.
+Unit tests are located in Mbed OS repository under `UNITTESTS` folder. Unit test files should use identical directory path to the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file under test is `rtos/Semaphore.cpp`, then all the test files should be in the `UNITTESTS/rtos/Semaphore` directory. Each test suite needs to have their own `unittest.cmake` file for test configuration.
 
-##### Test discovery
+#### Test discovery
 
 Registering unit tests for running is automatic, and the test runner handles registration. However, test files are not automatically assigned to be built. We build unit tests by using a separate build system, which searches for unit tests under the `UNITTESTS` directory.
 
 For the build system to find and build any test suite automatically, a unit test configuration file named `unittest.cmake` is required to be included with each unit test. This configuration file contains a name for the test and other source files the build requires.
```

Comment:
"is required to be included" - does that mean "you must include"? Because that's a much clearer phrasing.